planner/core/memtable_predicate_extractor.go

```diff
@@ -466,3 +477,92 @@ func (e *ClusterLogTableExtractor) Extract(
 	e.Patterns = patterns
 	return remained
 }
+
+// MetricTableExtractor is used to extract some predicates of metric_schema tables.
+type MetricTableExtractor struct {
+	extractHelper
+	// SkipRequest means the where clause always false, we don't need to request any component
+	SkipRequest bool
+	// StartTime represents the beginning time of metric data.
+	StartTime int64
+	// EndTime represents the ending time of metric data.
+	EndTime int64
+	// LabelConditions represents the label conditions of metric data.
+	LabelConditions map[string]set.StringSet
+	Quantiles       []float64
+}
+
+// Extract implements the MemTablePredicateExtractor Extract interface
+func (e *MetricTableExtractor) Extract(
+	ctx sessionctx.Context,
+	schema *expression.Schema,
+	names []*types.FieldName,
+	predicates []expression.Expression,
+) []expression.Expression {
+	// Extract the `quantile` columns
+	remained, skipRequest, quantileSet := e.extractCol(schema, names, predicates, "quantile", true)
+	e.Quantiles = e.parseQuantiles(quantileSet)
+	e.SkipRequest = skipRequest
+	if e.SkipRequest {
+		return nil
+	}
+
+	// Extract the `time` columns
+	remained, startTime, endTime := e.extractTimeRange(ctx, schema, names, remained, "time", ctx.GetSessionVars().StmtCtx.TimeZone)
+	if endTime == 0 {
+		endTime = math.MaxInt64
+	}
+	e.StartTime = startTime
+	e.EndTime = endTime
+	e.SkipRequest = startTime > endTime
```

Comment:
It's better to convert timestamp to time here instead of converting it in `GetQueryRangeTime` every time.

Reply:
done.